sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -288,8 +288,18 @@ private void CloseLink(ReceivingAmqpLink link)
             IEnumerable<string> lockTokens,
             TimeSpan timeout)
         {
-            Guid[] lockTokenGuids = lockTokens.Select(token => new Guid(token)).ToArray();
-            if (lockTokenGuids.Any(lockToken => _requestResponseLockedMessages.Contains(lockToken)))
+            bool requestResponse = false;
+            var lockTokenGuids = lockTokens.Select(token => new Guid(token)).ToArray();
+            foreach (var tokenGuid in lockTokenGuids)
+            {
+                if (_requestResponseLockedMessages.Contains(tokenGuid))
+                {
+                    requestResponse = true;
```

Comment:
You can inline the 

```
await DisposeMessageRequestResponseAsync();
return;

```

under the if check now, removing the extra variable.

Reply:
My preference was to keep the IO stuff out of the loop. I have no problems going your way if that is what you as maintainers prefer. Can only do in a few hours or I think you have edits rights on this PR

Reply:
I'll leave it to @ShivangiReja and @JoshLove-msft

Reply:
I don't have a strong preference. I will note that we already are following the pattern of returning from at multiple points in this method, which I find a bit harder to read, especially since we unconditionally call `DisposeMessagesAsync` at the end of the method. I would prefer if we had an else just to make it more clear.

Reply:
I like the way @pakrym suggested. If we would move `DisposeMessageRequestResponseAsync` inside the if block we can get rid of the extra variable.

Reply:
I can live with that.

Reply:
Fixed